src/Directory/Biobanks.Web/Controllers/BiobankController.cs

```diff
@@ -1004,6 +1004,32 @@ public async Task<ActionResult> GetAssociatedDataTypeViewsAjax(string id)
 
         }
 
+        private async Task<Boolean> linkedAssociatedDataIsValid(
+            List<AssociatedDataModel> linkedData, string ontologyTermId)
+        {
+            var associatedDataList = await _associatedDataTypeService.List();
+            var newAssociatedData = associatedDataList.Where(x => linkedData.Find(y=>y.DataTypeId == x.Id) != null);
+            // first check that all the data is present in the data list
+            if(newAssociatedData.Count() != linkedData.Count())
+            {
+                return false;
+            }
+            // then check that all the linked data is linked to the ontologyTerm
+            foreach (var type in newAssociatedData)
+            {
+                // only check linked data
+                if (type.OntologyTerms != null)
+                {
+                    if (type.OntologyTerms.Find(x => x.Id == ontologyTermId) == null)
```

Comment:
```suggestion
                if (type.OntologyTerms != null) && (type.OntologyTerms.Find(x => x.Id == ontologyTermId) == null)
```
could this be simplified to this?

Reply:
yes, and in fact even more idiomatic modern C# would be 
```suggestion
                if (type.OntologyTerms?.Find(x => x.Id == ontologyTermId) == null)
```

as the ["elvis operator"](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-) (`?.`) will shortcircuit the expression to `null` if `OntologyTerms` is null, instead of trying to run `Find()` on the `null` property.